<!DOCTYPE html>
<html>

<head>
<title>RNA-seq Differential Expression Analysis</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="How to use shell scripts in a unix environment to perform
RNA-seq differential expression analysis.">
<meta name="author" content="Harry Pollitt">
</head>
<body style="margin:30px 40px 30px 50px">
    <h1 style="font-size:30px;">RNA-Seq Analysis Protocol</h1>
	
    <p>This tutorial covers how to perform differential expression analysis using HISAT, HTSEQ and RNA. <br>
	Scripts can be found on github.com/Harry-Pollitt/RNA-Seq-Projects <br>
	I will show you each step, line by line and why I have done it that way </p>
	
	<p>Still to do... <br>
	- htseq feature counting <br>
	- getting your count matrices combined into one table <br>
	- loading that count table into R <br>
	- looking at differential expression with DESeq2 <br>
	- visualising differenetial expression results in R <br>
	- GO term enrichment and KEGG pathway analysis <br>
	
	<h1 style="font-size:20px;">Logging into ISCA</h1>
	
	<p>This may vary from what you've been taught. Just do what you normally do. When I connect I do so <br>
	from a shell command line using this command: </p>
	
	<div><pre><code class="language-shell">
	ssh user@login.isca.ex.ac.uk
	</code></pre></div>
	
	<p>Enter your normal password</p>
	
	<h1 style="font-size:20px;">Getting to your project folder</h1>
	
	<p>Find out where you are in the system. This should be your home directory.</p>
	<div><pre><code class="language-shell">
	pwd
	</code></pre></div>
	
	<p>Navigate to your project. Change TCODE for the required code.</p>
	
	<div><pre><code class="language-shell">
	cd /lustre/projects/Research_Project-TCODE
	</code></pre></div>
	
	<p>Should you need to return to your home directory use the command:</p>
	
	<div><pre><code class="language-shell">
	cd ~
	</code></pre></div>
	
	<p>Other useful commands using cd include: </p>
	
	<div><pre><code class="language-shell">
	cd ..     # move up one directory
	cd ../..  # move up two directories. Add more /.. to keep going
	cd -      # return to the last directory you were in
	</code></pre></div>
	
	<p>How you store your files is very important. A directory tree that makes sense will be useful <br>
	should you need to come back after some time. These are the commands to make directories. <br>
	Have a good think about the names of these. The should be simple but descriptive.</p>
	
	<div><pre><code class="language-shell">
	mkdir folder  
	mkdir -p folder/{subfolder1,subfolder2/{subfolder2-1,subfolder2-2}}
	</code></pre></div>
	
	<h1 style="font-size:20px;">Copying files to and within the HPC </h1>
		
	<p>Say you want to run a job in a particular directory but the script isn't there. <br>
	You can copy a script in two different ways. Either making a hard copy or by creating <br>
	what's called a symbolic link which tells the computer exactly where the script is kept. <br>
	A bit like a hyperlink. To do these use the following commands. </p>
	
	<div><pre><code class="language-shell">
	cp /path/to/script.sh where/I/want/it/script.sh  # copy a script to a different directory
	cp /path/to/source_directory /path/to/destination_directory  # you can copy entire directories
	ln -s /path/to/original-script.sh /path/to/linked-script.sh  # creates a link to the script
	</code></pre></div>
	
	<p>I recommend keeping your scripts in one directory to keep things neat and linking your <br>
	as needed. Slurm Workload Manager can also be used to choose which directory scripts are <br>
	ran in but that will be discussed later in the document.</p>
	
	<h1 style="font-size:20px;">Creating simple shell/bash scripts</h1>
	
	<p>You will often need to write commands straight into the commandline however commands longer <br>
	that a line or do something complicated should be writen to a file so that you have a record of <br>
	what you have done. This way you can reuse and easily modify them for another task. To create a <br>
	script file either make one outside of the HPC in a notepad and save as a .sh file or use this <br>
	command in the unix shell. </p>
	
	<div><pre><code class="language-shell">
	nano name-of-script.sh
	</code></pre></div>
	
	<p> This will open up a simple text editor that you can use to create scripts. Use arrow keys rather <br>
	than a mouse to navigate the file I would recommend using a dedicated editor such as Notepad++ <br>
	for longer scripts as it is easier to edit. </p>
	
	<h1 style="font-size:20px;">Running some Quality Control on your reads</h1>
	
	<p>Often sequencing companies will perform QC and trimming on your sequences before sending them to you <br>
    but it is good practice to take a look yourself to learn what it is they have done and why. There are a <br>
    couple of popular programs that are used for this task.</p>

    <p>FastQC is a tool that visualises the sequencing quality lines of your fastq file (the 4th line). It <br>
    also checks for adaptor sequences should be disgarded. It will produce several graphs that will highlight <br>
    any problems your data might have. A section maked as failed may not necessarily mean that the reads are <br>
    poor quality. This program can be used with DNA and RNA which have different QC criteria. You should check <br>
    <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">the documentation</a> for more details. </p>	
	
	<h1 style="font-size:20px;">Creating an Reference Genome Index using HISAT2</h1>
	
	<p>Sequence aligners usually require the reference genome to be "Indexed" before it can be used. <br>
	Running the index builder will create several files that will be used later on to align the <br>
	sequencing reads to the reference. This only needs to be done once and can be reused. <br>
	This will need to be done again should a better reference genome be made. Just make <br>
	a note of which one you used. </p>
	
	<p> We'll be using HISAT2 to perform out indexing. Other aligners such as STAR will also work. <br>
	It is important for aligning RNA that the aligner is splice aware. Don't just use any aligner. <br>
	You can find more information about HISAT2 <a href="http://daehwankimlab.github.io/hisat2/manual/">here</a> 
	and the reference <a href="https://doi.org/10.1038/s41587-019-0201-4">here</a>. <br>
	To Index the reference genome. Download the fasta file and annotation file in GTF format and <br>
    store them in the Reference directory.	</p>
	
	<p> Next we'll submit the hisat2-index.sh script to the Slurm Workload Manager. While in the <br>
	Reference directory run this command. </p>
	
	<div><pre><code class="language-shell">
	sbatch hisat2-index.sh
	</code></pre></div>
	
	<p>You can check if the job is running by entering the command </p>
	
	<div><pre><code class="language-shell">
	squeue -u USERID
	</code></pre></div>
	
	<p>Once the job has finished running check the Reference directory to see if the correct files were created. <br>
	You can do this using the ls command.</p>
	
	<div><pre><code class="language-shell">
	ls -lh  # -lh is a human readable longer format
	</code></pre></div>
	
	<p>The '-l' option for 'ls' gives a bit more detail for each file. It will tell you when the permissions <br>
	of the file, who created it, the size and when it was created. The '-h' option will convert the size to <br>
	a human readable version. It will abreviate large numbers to kb/mb/gb to make it easier to read.</p>
	
	<h1 style="font-size:20px;">Aligning your reads to a reference genome using HISAT2</h1>
	
	<p>By aligning your reads to a reference genome you can see how much a gene has been expressed. More reads <br> 
	equals higher expression. Before we can run the alignment with all our samples it's good to know roughly how <br>
	long each alignment will take so we can ask for the correct amount of time for the job with Slurm. <br>
	Run the following script. First checking and editing the SBATCH lines within the script. Change the <br> 
	Time, threads and directory pathways as required. Hint: use nano </p>
	
	<div><pre><code class="language-shell">
	sbatch hisat2-test-alignment.sh read1.fq read2.fq
	sacct --format="Elapsed" -j <myid>  # See how much time was needed for the job
	</code></pre></div>
	
	<p>Once you have an idea of how long each alignment is going to take you can edit the next script to ask <br>
	for the correct required time. Then we can align all the samples in one go. All you need to do is make sure <br> 
	the script is in the correct folder (the one with the reads.fastq), check all the SBATCH lines are correct <br>
	and submit the job.</p>
	
	<div><pre><code class="language-shell">
	sbatch hisat2-alignment.sh 
	</code></pre></div>
	
</body>
</html>